diff --git a/src/Console/Command/Compile.php b/src/Console/Command/Compile.php
index ebc1ee6..6806d26 100644
--- a/src/Console/Command/Compile.php
+++ b/src/Console/Command/Compile.php
@@ -18,6 +18,7 @@ use Amp\MultiReasonException;
 use function array_map;
 use function array_search;
 use function array_shift;
+use function Composer\Autoload\includeFile;
 use function count;
 use function decoct;
 use function explode;
@@ -96,6 +97,7 @@ HELP;
     private const DEV_OPTION = 'dev';
     private const NO_CONFIG_OPTION = 'no-config';
     private const WITH_DOCKER_OPTION = 'with-docker';
+    private const BOOTSTRAP_OPTION = 'bootstrap';
 
     private const DEBUG_DIR = '.box_dump';
 
@@ -146,6 +148,12 @@ HELP;
             InputOption::VALUE_NONE,
             'Generates a Dockerfile'
         );
+        $this->addOption(
+            self::BOOTSTRAP_OPTION,
+            null,
+            InputOption::VALUE_REQUIRED,
+            'A PHP script that is included before compile run',
+        );
 
         $this->configureWorkingDirOption();
     }
@@ -174,6 +182,14 @@ HELP;
 
         $this->changeWorkingDirectory($input);
 
+        if (!empty($bootstrap = $input->getOption(self::BOOTSTRAP_OPTION)) && file_exists($bootstrap)) {
+            $io->writeln(
+                sprintf('<info>[debug] Bootstrapped file "%s"</info>', $bootstrap),
+                OutputInterface::VERBOSITY_DEBUG,
+            );
+            includeFile($bootstrap);
+        }
+
         $io->writeln($this->getApplication()->getHelp());
         $io->newLine();
 
@@ -546,7 +562,15 @@ HELP;
             );
 
             if (is_callable($metadata)) {
-                $metadata = $metadata();
+                $logger->log(
+                    CompilerLogger::MINUS_PREFIX,
+                    'Using composer.json : ' . $config->getComposerJson()
+                );
+                $logger->log(
+                    CompilerLogger::MINUS_PREFIX,
+                    'Using composer.lock : ' . $config->getComposerLock()
+                );
+                $metadata = $metadata($config, $box);
             }
 
             $logger->log(
